### PR TITLE
nnet3: removing add_lda and include_log_softmax options

### PR DESCRIPTION
**Pull request by vimalmanohar**

### CODE REVIEW

**Comment by danpovey** [comment]
Thanks a lot... I'm afraid I am working on a bunch of other changes that may conflict with this so, so I think it's better if we wait and have you commit this in a day or two.  I probably won't need the lda-related code as I think already resolved that issue everywhere, but the other part is relevant.

**Comment by danpovey** [comment]
OK I have committed those other changes I was talking about now.  you'll probably have conflicts.

**Comment by danpovey** [comment]
@vimalmanohar, can you please try to resolve the conflicts in this PR?

**Comment by danpovey** [comment]
@vimalmanohar, sorry, I dropped the ball on this, and new conflicts have appeared.  Would you mind resolving them?

File: egs/wsj/s5/steps/nnet3/train_raw_dnn.py [hunk 1]
```diff
@@ -286,7 +288,9 @@ def train(args, run_opts):
     # use during decoding
     common_train_lib.copy_egs_properties_to_exp_dir(egs_dir, args.dir)
 
-    if (args.stage <= -3) and os.path.exists(args.dir+"/configs/init.config"):
```
Comment (danpovey) [inline egs/wsj/s5/steps/nnet3/train_raw_dnn.py]: Are there other reasons we use init.config other than to make the LDA-like transform?

File: egs/wsj/s5/steps/nnet3/train_raw_dnn.py [hunk 2]
```diff
@@ -286,7 +288,9 @@ def train(args, run_opts):
...
+    add_lda = common_train_lib.is_lda_added(config_dir)
+
```
Comment (danpovey) [inline egs/wsj/s5/steps/nnet3/train_raw_dnn.py]: Are there other reasons we use init.config other than to make the LDA-like transform?

Comment (vimalmanohar) [inline egs/wsj/s5/steps/nnet3/train_raw_dnn.py]: Probably not. It seems like to be removed in the transfer learning PR.

**Comment by danpovey** [comment]
In that case can we just keep the current code instead of adding a new
function to check it?

File: egs/wsj/s5/steps/nnet3/train_raw_dnn.py (hunk 2 above)
Comment (vimalmanohar) [inline egs/wsj/s5/steps/nnet3/train_raw_dnn.py]: init.config is still used to create the initial model. The check is needed to know if the LDA needs to be trained.

**Comment by danpovey** [comment]
I think you are mistaken, if we are talking about the current kaldi_52
code; init.config is only used if we are doing the LDA thing.

File: egs/wsj/s5/steps/nnet3/train_raw_dnn.py (hunk 2 above)
Comment (danpovey) [inline egs/wsj/s5/steps/nnet3/train_raw_dnn.py]: did you see my comment:
"I think you are mistaken, if we are talking about the current kaldi_52
code; init.config is only used if we are doing the LDA thing."
can you please update the PR?

Comment (vimalmanohar) [inline egs/wsj/s5/steps/nnet3/train_raw_dnn.py]: I do not understand what needs to be done. The current solution is needed to know if LDA needs to be trained. The function is_lda_added can be changed to read init.raw if needed.

**Comment by danpovey** [comment]
Look at the current xconfig_to_configs.py in kaldi_52: it skips creating
the init.config if there are no output nodes in init.config, and the only
layer that adds an output node there (I believe) is the LDA layer.  So I
believe the script only needs to check for the existence of the
init.config.  the add_lda variable is not needed at all.

**Comment by danpovey** [comment]
Closing this as I'm working on a version of this myself, along with certain other code cleanups.

**Pull request closed without merging**